### PR TITLE
wrapping of wallet endpoints for motor in wasm compatibility

### PR DESCRIPTION
**Pull request by joshLong145**
-- addition of wasm wrapper for motor wallet endpoints

## Changes
* Item 1
* Item 2

## Checklist
* [ ] Unit tests
* [ ] Documentation

## References _(optional)_

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202528620406598/1202529466067520) by [Unito](https://www.unito.io)

### CODE REVIEW

File: bind/motor-wasm/wasm.go [hunk 1]
```diff
@@ -1,18 +1,263 @@
...
+}
+
+// Address returns the address of the wallet.
+func Address() string {
+	if instance == nil {
+		return ""
+	}
+	addr, err := instance.wallet.Address()
+	if err != nil {
+		return ""
```
Comment (ntindle) [inline bind/motor-wasm/wasm.go]: Should we be returning the errors in some way instead of swallowing them

File: bind/motor-wasm/wasm.go [hunk 2]
```diff
@@ -1,18 +1,263 @@
...
+}
+
+// MarshalWallet returns the JSON representation of the wallet.
+func MarshalWallet() []byte {
+	if instance == nil {
+		return nil
+	}
+	buf, err := instance.wallet.Marshal()
+	if err != nil {
+		return nil
```
Comment (ntindle) [inline bind/motor-wasm/wasm.go]: Why nil over an empty array?

File: bind/motor-wasm/wasm.go [hunk 3]
```diff
@@ -1,18 +1,263 @@
...
+	}
+	return buf
+}
+
+// Sign returns the signature of the given message.
+func Sign(typeUrl string, msg []byte) ([]byte, error) {
+	if instance == nil {
+		return nil, errWalletNotExists
```
Comment (ntindle) [inline bind/motor-wasm/wasm.go:127]: This deviates from our paradigm above of not retuning two params

File: bind/motor-wasm/wasm.go [hunk 4]
```diff
@@ -1,18 +1,263 @@
...
+func Verify(msg []byte, sig []byte) bool {
+	if instance == nil {
+		return false
+	}
+	return instance.wallet.Verify(msg, sig)
+}
+
+// Broadcast broadcasts rawTx to the specified address
+func Broadcast(addr string, tx []byte) error {
+	apiEndpoint := "v1-beta.sonr.ws:1317/cosmos/tx/v1beta/txs"
```
Comment (ntindle) [inline bind/motor-wasm/wasm.go]: This worries me but if it must be somewhere can we consolidate it to one central location?

**Review by ntindle — CHANGES_REQUESTED** [review]

File: bind/motor-wasm/wasm.go (hunk 1 above)
Comment (mcjcloud) [reply bind/motor-wasm/wasm.go]: +1, the binding should support errors. If wasm only supports single return values, we could return arrays that represent tuples.

File: bind/motor-wasm/wasm.go (hunk 2 above)
Comment (mcjcloud) [reply bind/motor-wasm/wasm.go]: I think nil makes sense here actually. The idea won't be to operate on the contents of the array but just to deserialize it. So it's kind of an all-or-nothing sort of array.

**Review by mcjcloud — CHANGES_REQUESTED** [review]
Consider making protos for requests/responses of these functions, since that's how target code will be interacting with go code.

File: bind/motor-wasm/wasm.go (hunk 4 above)
Comment (joshLong145) [reply bind/motor-wasm/wasm.go]: yea we shouldnt be statically defining endpoints. this was taken from the `bind.go` implementations.

File: bind/motor-wasm/wasm.go (hunk 1 above)
Comment (joshLong145) [reply bind/motor-wasm/wasm.go]: good call

File: bind/motor-wasm/wasm.go (hunk 3 above)
Comment (joshLong145) [reply bind/motor-wasm/wasm.go:127]: the wrapper for the language target will filter to single value returns.

File: bind/motor-wasm/wasm.go (hunk 4 above)
Comment (joshLong145) [reply bind/motor-wasm/wasm.go]: moved to a variable

**Review by prdlk — APPROVED** [review]

**Pull request merged**